File: ui/src/app/Product.ts
export interface Product {
    id: number; 
    name: string;
    internalName: string;
    details: string;
    maxProductsPerLocation: number;
}

export interface Feature{
    id: number;
    name: string;
    internalName: string;
    details: string;
    productName: string;
    productInternalName: string;
}

export interface Parameter{
    id: number;
    name: string;
    internalName: string;
    details: string;
    parameterType: ParameterType;
    value: string
}
 
export enum ParameterType {
    QUANTITY = 'QUANTITY',
    PRICE = 'PRICE',
    OTHER = 'OTHER',
}
  

File: bakend/src/main/java/com/learnings/capstone/entity/Product.java
package com.learnings.capstone.entity;


import lombok.Getter;
import lombok.Setter;
import jakarta.persistence.*;
import java.util.List;

@Entity
@Table(name = "products")
@Getter
@Setter
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "internal_name")
    private String internalName;

    @Column(name = "details")
    private String details;
    
    @JoinColumn(name = "user_id")
    private Long userId;

    @Column(name = "max_products_per_location")
    private Integer maxProductsPerLocation;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<Feature> features;

	
    public Product() {
    }

    public Product(Long id, String name, String internalName, String details, Integer maxProductsPerLocation) {
        this.id = id;
        this.name = name;
        this.internalName = internalName;
        this.details = details;
        this.maxProductsPerLocation = maxProductsPerLocation;
    }
    
}
